### PR TITLE
Add keybinding to toggle selected layer visibility

### LINKED ISSUE #1139: Toggle current layer visibility w/keyboard shortcut
## 🚀 Feature

I'd like to introduce a new keyboard shortcut to toggle visibility of current layer

## Motivation

When working with a few layers, I often find myself wanting to toggle the visibility of a particular layer to look clearly at one beneath. This has been coming up most recently when painting labels.

## Pitch

I'd like to propose a new keyboard shortcut to toggle the visibility of the current layer. 

This part is up for debate, but if this is very useful for people, it could be a single key, like `H`, or a combo like `Ctrl+H`, or some other key(s) entirely. 

From looking at the other key bindings, I assume this would be straightforward to do, unless there is some `Qt` issue I'm not familiar with.

## Alternatives

At the moment, I use the mouse to manually toggle the visibility.

### PR DESCRIPTION
**Pull request by hectormz**
# Description

This adds a function to toggle selected layer visibility along with a key binding (`Ctl+H`)

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# References

This PR (partially) closes #1139

# How has this been tested?

- [X] Added test to select some of layers with heterogeneous visibility and toggle the visibility back and forth

## Final checklist:
- [X] My PR is the minimum possible work for the desired functionality
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by tlambert03** [comment]
just a quick note: `Control-H` (on a mac at least, and I thought also windows?) is reserved for hiding the entire napari application.  So we'll probably need to pick a slightly different combo.
("control" here actually maps to the "command" key on macs)

**Comment by hectormz** [comment]
@tlambert03 Good to know, I wasn't aware of that. That's not the case on Windows (I think). From my own laziness, I think a one or two key shortcut works well. How do you feel about just `H`?

**Comment by tlambert03** [comment]
I personally kinda like hotkeys without any modifiers, like just `H`, however those can be complicated in programs that also accept text (imagine you're editing the name of a layer and suddenly it disappears).  It's probably doable, but I believe that would be our first un-modified hotkey... so we'd be wading into new territory with lots of testing to do.

**Comment by hectormz** [comment]
Good point. I didn't think about that. I did a really quick test and changed it to `H`. Nothing was triggered when editing layer names and typing `H` or in file dialog, but I'm not sure all the corner cases that `napari` might have.

Other options might be `Ctrl-V` (for visible) or `Ctrl-I` (for invisible), but I think those are better served with potential pasting, or showing layer info

**Comment by tlambert03** [comment]
keypress events are handled by the active widget, and then (sometimes) [cascade up through the parent widgets](https://doc.qt.io/qt-5/eventsandfilters.html#event-handlers), depending on how the specific widget and its key event methods and event filters were implemented.  So I'm not surprised it worked on the LayerList widget... but like you say, we'd need to think about every widget (or someone would need to read up on event filters and figure out whether we're safe to have it on the main viewer).

I also like short key combos ... but unfortunately there are only so many letters we get :) i agree that ctrl-v and i are better served by paste and info.  so might need to go with an unintuitive letter, or add another modifier (like ctrl-shift-V)

**Comment by hectormz** [comment]
@tlambert03 I agree. May be there should be a roadmap for future keybindings so we don't waste a good one now 😂. My last thought unless others have an idea is `Ctrl-D` for `Display/Hide`

**Comment by tlambert03** [comment]
What about alt instead of ctrl?  Alt-V?

**Comment by jni** [comment]
I'm +1 on unmodified shortcuts, and advocated a lot for them early on. Alt is annoying, not many commands (or any?) work that way on Mac, without Cmd also pressed.

Since visibility is easy to revert, I don't think it's a huge risk to accidentally hide a layer. It's obvious in the UI what has happened, and easy to toggle back.

**Comment by sofroniewn** [comment]
After discussion with @jni, @tlambert03 we propose to just use the unmodified `V` key. If you make the change @hectormz then this PR should be good to go

You can see all the keybindings we currently have in the `Show key bindings` dialog in the help menu.

**Comment by hectormz** [comment]
@sofroniewn I updated to `V`. Thanks to @tlambert03  and @jni for weighing in

File: napari/components/layerlist.py [hunk 1]
```diff
@@ -173,3 +173,9 @@ def select_previous(self, shift=False):
                 self[selected[0] - 1].selected = True
         elif len(self) > 0:
             self[0].selected = True
+
+    def toggle_selected_visibility(self):
+        """Toggle visibility of selected layers"""
+        for i in range(len(self)):
+            if self[i].selected:
+                self[i].visible = not self[i].visible
```
Comment (sofroniewn) [inline napari/components/layerlist.py]:
```suggestion
        for layer in self:
            if layer.selected:
                layer.visible = not layer.visible
```
Minor request to change, but because this class subclasses a list you can actually just iterate right through it, making the code slightly more readable

Comment (hectormz) [reply napari/components/layerlist.py]: 👌 good point!

**Review by sofroniewn — APPROVED** [review]
Looks great to me!! Thanks @hectormz

**Review by jni — APPROVED** [review]

**Review by tlambert03 — APPROVED** [review]

**Comment by jni** [comment]
Just played with this locally and oh man, I love it! 😃😃😃😃 Thanks @hectormz !

**Comment by hectormz** [comment]
@jni awesome! `napari` made it very easy to add the functionality. I've been enjoying it more than I would too!
